core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -204,4 +226,190 @@ public Iterable<FileScanTask> split(long splitSize) {
       return ImmutableList.of(this); // don't split
     }
   }
+
+  static StaticDataTask.Row manifestFileToRow(PartitionSpec spec, ManifestFile manifest, long referenceSnapshotId) {
+    return StaticDataTask.Row.of(
+        manifest.content().id(),
+        manifest.path(),
+        manifest.length(),
+        manifest.partitionSpecId(),
+        manifest.snapshotId(),
+        manifest.content() == ManifestContent.DATA ? manifest.addedFilesCount() : 0,
```

Comment:
I think 0 may be the wrong thing here, shouldn't it be "null" if we don't have the info?

Reply:
I think the problem is in the original schema in `ManifestsTable`, where the counts are required. I also did not notice that data file counts are optional in `AllManifestsTable` when I recently added counts for delete files. I think we should continue to use 0 for compatibility but I wonder why we have such a mismatch in nullability between these tables.

Reply:
I'd still use 0 for data files count when reading a delete manifest as null means unknown but we know there is no data file in a delete manifest.

Reply:
Yea this is copy + add from ManifestsTable class (should have mentioned), so if we fix it can be in another pr.